Assets/Scripts/Puzzle/ItemPlace.cs

```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+using System;
+public class ItemPlace : MonoBehaviour
+{
+    [SerializeField] string _itemName;
+    [SerializeField] private InventoryUI _inventoryUI;
+    [SerializeField] private ItemPuzzle _itemPuzzle;
+    [SerializeField] private LayerMask _layrMask;
+    
+
+    public event Action OnAdd;
```

Comment:
rename this to OnItemAdded or OnItemPlaced, OnAdd does not give enough information